app/src/main/java/com/example/masterproject/ledger/Ledger.kt

```diff
@@ -59,6 +59,23 @@ class Ledger {
             return null
         }
 
+        /**
+         * @description: Return this users entry in the existing ledger if the user
+         *                has a stored certificate that is part of the ledger.
+         *                Returns null if not.
+         * */
+        fun existingLedgerEntryFromStoredCertificate(): LedgerEntry? {
+            val context = App.getAppContext()
+            if (context != null) {
+                val storedCertificate = PKIUtils.fetchStoredCertificate(context)
```

Comment:
Litt usikker på om det kan bli en case, men også om det finnes et sertificat i minne som ikke er lagret i filsystemet. 

```suggestion
                val storedCertificate = PKIUtils.fetcheStoredCertifiacate(context) ?: PKUUtils.getCertificate()
```